hack/jenkins/e2e.sh

```diff
@@ -190,6 +204,194 @@ case ${JOB_NAME} in
+
+  # GCE master+node upgrade step 1: deploy a cluster at latest release.
+  kubernetes-upgrade-gce-step1-deploy)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_DOWN:="false"}
+    : ${E2E_TEST:="false"}
+    : ${JENKINS_USE_RELEASE_TARS:="y"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GCE master+node upgrade step 2: upgrade the cluster from step 1 to the
+  # latest CI version using the latest CI verion's upgrade code.
+  kubernetes-upgrade-gce-step2-upgrade)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_DOWN:="false"}
+    : ${E2E_OPT:="--check_version_skew=false"}
+    : ${E2E_UP:="false"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Skipped.*Cluster\supgrade.*upgrade-cluster"}
+    : ${JENKINS_FORCE_GET_TARS:="y"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GCE master+node upgrade step 3: run the standard set of non-flaky tests,
+  # using the tests at the version of the code retrieved in step 2.
+  kubernetes-upgrade-gce-step3-e2e)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_OPT:="--check_version_skew=false"}
+    : ${E2E_UP:="false"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GKE
+  # ---
+
+  # Launch a GKE cluster at latest CI, run all non-flaky tests sequentially.
+  kubernetes-e2e-gke-ci)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-ci"}
+    : ${E2E_NETWORK:="e2e-gke-ci"}
+    : ${E2E_SET_CLUSTER_API_VERSION:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    ;;
+
+  # Launch a GKE cluster at latest CI, run the reboot tests and the non-flaky
+  # tests sequentially.
+  kubernetes-e2e-gke-ci-reboot)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jenkins-gke-e2e-ci-reboot"}
+    : ${E2E_NETWORK:="e2e-gke-ci"}
+    : ${E2E_SET_CLUSTER_API_VERSION:="y"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_SKIP_KEEP_REBOOT_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${PROJECT:="k8s-jkns-e2e-gke-ci"}
+    ;;
+
+  # Launch a GKE cluster in test, run non-flaky tests sequentially.
+  kubernetes-e2e-gke-test)
+    : ${CMD_GROUP:="beta"}
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-test"}
+    : ${E2E_NETWORK:="e2e-gke-test"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${JENKINS_USE_RELEASE_TARS:="y"}
+    # TODO(mbforbes): Do we need to set PATH here?
+    : ${PATH="/usr/local/share/google/google-cloud-sdk/bin:${PATH}"}
```

Comment:
tl;dr, not needed here.

`PATH` is overwritten to set which version of gcloud/kubectl gets used. However, looks like [ginkgo-e2e.sh](https://github.com/mbforbes/kubernetes/blob/gkeJenkins/hack/ginkgo-e2e.sh#L37) sets the kubectl version via an envvar now, so overwriting path when we are using default gcloud isn't needed. Instead we should probably be setting
```
${KUBECTL:="/usr/local/share/google/google-cloud-sdk/bin/kubectl"}
```